customcontext/context.h

```diff
@@ -59,22 +59,54 @@
 #include "threaduploadtexture.h"
 #endif
 
+#if QT_VERSION >= 0x050800
+#include <private/qsgdefaultcontext_p.h>
+#include <private/qsgdefaultrendercontext_p.h>
+#endif
 
 #if QT_VERSION >= 0x050200
 struct QSGMaterialType;
 #endif
 
+#if QT_VERSION >= 0x050800
+#define CONTEXT_CLASS Context
+#define CONTEXT_CLASS_BASE QSGDefaultContext
+#define RENDER_CONTEXT_CLASS RenderContext
+#define RENDER_CONTEXT_CLASS_BASE QSGDefaultRenderContext
+#define COMPILESHADER_METHOD compileShader
+#elif QT_VERSION >= 0x050200
```

Comment:
Don't think we need anymore support for below 5.2. By this point could just make this #else and drop the last set. Might allow to drop some definitions maybe?

Reply:
In our Qt 6 builds we've dropped it completely for now. The rework required was quite big and seems we can get away with not having the whole qtscenegraph-adaptation for now by the looks of it.

Reply:
> Don't think we need anymore support for below 5.2. By this point could just make this #else and drop the last set. Might allow to drop some definitions maybe?

Happy for suggestions there, just wanted to upstream some of our patches, so they would be useful for you guys down the line and we don't need to fork and keep carrying them :)